fvm/meter/interaction_meter.go

```diff
@@ -51,34 +59,68 @@ func (m *InteractionMeter) MeterStorageRead(
 ) error {
 
 	// all reads are on a View which only read from storage at the first read of a given key
-	if _, ok := m.storageUpdateSizeMap[storageKey]; !ok {
+	if _, ok := m.reads[storageKey]; !ok {
 		readByteSize := getStorageKeyValueSize(storageKey, value)
 		m.totalStorageBytesRead += readByteSize
-		m.storageUpdateSizeMap[storageKey] = readByteSize
+		m.reads[storageKey] = readByteSize
 	}
 
 	return m.checkStorageInteractionLimit(enforceLimit)
 }
 
 // MeterStorageWrite captures storage written bytes count and returns an error
-// if it goes beyond the total interaction limit and limit is enforced
+// if it goes beyond the total interaction limit and limit is enforced.
+// If a key is written multiple times, only the last write is counted.
+// If a key is written before it has been read, next time it will be read it will be from the view,
+// not from storage, so count it as read 0.
 func (m *InteractionMeter) MeterStorageWrite(
 	storageKey flow.RegisterID,
 	value flow.RegisterValue,
 	enforceLimit bool,
 ) error {
-	// all writes are on a View which only writes the latest updated value to storage at commit
-	if old, ok := m.storageUpdateSizeMap[storageKey]; ok {
-		m.totalStorageBytesWritten -= old
-	}
-
 	updateSize := getStorageKeyValueSize(storageKey, value)
-	m.totalStorageBytesWritten += updateSize
-	m.storageUpdateSizeMap[storageKey] = updateSize
+	m.replaceWrite(storageKey, updateSize)
+
+	if _, ok := m.reads[storageKey]; !ok {
+		// write without read, count as read 0 because next time you read it the written value will be returned
+		// so no interaction with storage will happen.
```

Comment:
```suggestion
		// write without read, count as read 0 because next time you read it the written value 
		// from cache will be returned, so no interaction with storage will happen.
```